djangoProject3/my-bookstore/src/components/CartContext.js

```diff
@@ -0,0 +1,102 @@
+import { createContext, useState } from "react";
+import {  getProductData } from "./productsStore";
+
+
+export const CartContext = createContext({
+    items: [],
+    getProductQuantity: () => {},
+    addOneToCart: () => {},
+    removeOneFromCart: () => {},
+    deleteFromCart: () => {},
+    getTotalCost: () => {},
+});
+
+export function CartProvider({children}) {
+    const [cartProducts, setCartProducts] = useState([]);
+    
+    function getProductQuantity(id) {
+        const quantity = cartProducts.find(product => product.id === id)?.quantity;
+        
+        if (quantity === undefined) {
+            return 0;
+        }
+
+        return quantity;
+    }
+
+    function addOneToCart(id) {
+        const quantity = getProductQuantity(id);
+        if (quantity === 0) {
+            setCartProducts(
+                [
+                    ...cartProducts,
+                    {
+                        id: id,
+                        quantity: 1,
+                    }
+                ]
+            )
+        } else {
+            setCartProducts(
+                cartProducts.map(
+                    product =>
```

Comment:
the better way to set state if it is dependent on a previous state use following

```setCounter(prev=> prev+1);```

you can name state anything as I have prev, but from use state hook we may have restructured it as ```counter```